Kernel/Tasks/PowerManagementTask.cpp

```diff
@@ -18,46 +19,123 @@
 #include <Kernel/FileSystem/VirtualFileSystem.h>
 #include <Kernel/Firmware/ACPI/Parser.h>
 #include <Kernel/Library/Panic.h>
+#include <Kernel/Locking/WritableOnce.h>
 #include <Kernel/Sections.h>
 #include <Kernel/Tasks/FinalizerTask.h>
-#include <Kernel/Tasks/PowerStateSwitchTask.h>
+#include <Kernel/Tasks/PowerManagementTask.h>
 #include <Kernel/Tasks/Process.h>
 #include <Kernel/Tasks/Scheduler.h>
+#include <Kernel/Tasks/WaitQueue.h>
 
 namespace Kernel {
 
-Thread* g_power_state_switch_task;
+enum class PowerStateCommand {
+    None,
+    Shutdown,
+    Reboot,
+};
+
+static constexpr StringView power_management_task_name = "Power Management Task"sv;
+Thread* g_power_management_task_thread;
+WritableOnce<PowerStateCommand> g_power_management_task_command { PowerStateCommand::None };
```

Comment:
Is this truly Writeable once? I could foresee us wanting to do suspend/resume as well in the future.

Reply:
Sure, but for now we don't support these things, so maybe a FIXME is sufficient for now?

Reply:
I have something in place to fix this in a better way :)